javascript-closures/my-app/src/Counter.tsx

```diff
@@ -0,0 +1,16 @@
+import { useCallback, useEffect, useState } from 'react';
+
+export function Counter() {
+  const [counter, setCounter] = useState(0);
+  const data = useCallback(function getData() {
+    // fetch data
+    return { foo: 'bar' };
+  }, []);
+
+  useEffect(() => {
+    data;
+    setCounter((prev) => prev + 1);
+  }, [data, setCounter]);
```

Comment:
You don't need to put `setCounter` in here since React guarantees that the function returned by `useState` will never change. The lint rule knows this, also.